tests/compatibility-tests/src/main/resources/serial/serial.groovy

```diff
@@ -24,30 +24,40 @@ import org.apache.activemq.artemis.jms.client.*
 
 file = arg[0]
 method = arg[1]
-System.out.println("File::" + file);
+version = arg[2]
+System.out.println("File::" + file)
 
 
 if (method.equals("write")) {
     cf = new ActiveMQConnectionFactory("tcp://localhost:61616?confirmationWindowSize=1048576&blockOnDurableSend=false");
     queue = new ActiveMQQueue("queue");
     topic = new ActiveMQTopic("topic")
+    if (version.equals("ARTEMIS-SNAPSHOT")) {
+        destination = new ActiveMQDestination("address", "name", ActiveMQDestination.TYPE.DESTINATION, null)
```

Comment:
I dont think we should be creating ActiveMQDestination with "name" in 2.X this constructor should be deprecated, the "name" is only for compatibility with older client version.

Reply:
Verifying compatibility is exactly what this test is for so it's valid for it to be using the deprecated fields.

Reply:
Fair enough, you sold me.